### PR TITLE
[jit][x86][cxx] int/ptr casts in mini-x86.c

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/mini/mini-x86.c [hunk 1]
```diff
@@ -3468,7 +3468,7 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 				x86_fld80_membase (code, X86_ESP, 2);
 				x86_alu_reg_imm (code, X86_ADD, X86_ESP, 12);
 			} else {
-				x86_fld80_mem (code, mn);
+				x86_fld80_mem (code, (gsize)&mn);
```
Comment (lewurm) [inline mono/mini/mini-x86.c]: shouldn't it be `(gsize)mn`?

Comment (jaykrell) [reply mono/mini/mini-x86.c]: Same thing. I thought this might be clearer, but same thing.

Comment (lewurm) [reply mono/mini/mini-x86.c]: ah arrays. nevermind 🙂

Comment (jaykrell) [reply mono/mini/mini-x86.c]: Could probably be `const` too.

**Comment by jaykrell** [comment]
@monojenkins squash

**Pull request merged**